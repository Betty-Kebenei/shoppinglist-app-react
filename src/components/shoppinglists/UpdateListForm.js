import React, { Component } from 'react';
import instance from '../AxiosInstance';
import toastr from 'toastr';
import { 
    Modal,
    FormGroup, 
    ControlLabel,
    FormControl,
    Button 
} from 'react-bootstrap';

class UpdateListForm extends Component {
    constructor(props) {
      super(props);
  
      this.handleChange = this.handleChange.bind(this);
  
      this.state = {
        listname: props.listName,
      };
    }

    handleChange = (e) => {
        this.setState({ [e.target.name]: e.target.value });
    }

    handleSubmit = (event) => {
        event.preventDefault();
        let values = new FormData();
        values.set("listname", this.state.listname);

        const { listId } = this.props;

        instance.put(`/shoppinglists/${listId}`, values).then(
            response => {
                this.setState({
                    showModal4: false 
                })
                this.props.getAllShoppingLists();
                toastr.success("You have successfully updated a shopping lists!")
            }).catch(error =>{
                    toastr.error(error.response.data.message)
            })
    }
  
    render() {
      return (
            <div>
                <button 
                    type="button" 
                    className="btn glyphicon glyphicon-edit text-primary"
                    onClick={() => this.setState({showModal4:true})} 
                    data-toggle="tooltip" 
                    data-placement="top" 
                    title="Update_List" 
                    />
                <Modal show={this.state.showModal4} onHide={() => this.setState({ showModal4:false})}>
                    <Modal.Header closeButton>
                        <Modal.Title>Edit Shopping List</Modal.Title>
                    </Modal.Header>
                    <Modal.Body>
                        <form onSubmit={this.handleSubmit}>
                            <FormGroup
                                controlId="Listname"
                                >
                                <ControlLabel>Listname</ControlLabel>
                                    <FormControl
                                        type="text"
                                        name="listname"
                                        value={this.state.listname}
                                        placeholder="Enter listname"
                                        onChange={this.handleChange}
                                        required
                                    />
                            </FormGroup>
                            <Button type="submit">Submit</Button>
                            <FormControl.Feedback />
                        </form>  
                        </Modal.Body>
                    <Modal.Footer>
                        <Button onClick={() => this.setState({ showModal4:false})}>Close</Button>
                    </Modal.Footer>
                </Modal>
            </div>
      );
    }
  }
  
export default UpdateListForm;