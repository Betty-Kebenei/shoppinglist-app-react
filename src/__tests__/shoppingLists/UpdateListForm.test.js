import React from 'react';
import { shallow, mount, render } from 'enzyme';

import Config from '../../Config';

import UpdateListForm from '../../components/shoppinglists/UpdateListForm';

describe('interface of UpdateListForm component', () => {
    it('should render', () => {
        const wrapper = shallow(<UpdateListForm />);
        expect(wrapper).toMatchSnapshot();
    });
});