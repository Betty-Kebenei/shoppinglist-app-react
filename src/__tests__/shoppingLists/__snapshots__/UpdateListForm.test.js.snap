// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`interface of UpdateListForm component should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UpdateListForm />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          className="btn glyphicon glyphicon-edit text-primary"
          data-placement="top"
          data-toggle="tooltip"
          onClick={[Function]}
          title="Update_List"
          type="button"
        />,
        <Modal
          animation={true}
          autoFocus={true}
          backdrop={true}
          bsClass="modal"
          dialogComponentClass={[Function]}
          enforceFocus={true}
          keyboard={true}
          manager={
            ModalManager {
              "add": [Function],
              "containers": Array [],
              "data": Array [],
              "handleContainerOverflow": true,
              "hideSiblingNodes": true,
              "isTopModal": [Function],
              "modals": Array [],
              "remove": [Function],
            }
          }
          onHide={[Function]}
          renderBackdrop={[Function]}
          restoreFocus={true}
          show={false}
        >
          <ModalHeader
            bsClass="modal-header"
            closeButton={true}
            closeLabel="Close"
          >
            <ModalTitle
              bsClass="modal-title"
              componentClass="h4"
            >
              Edit Shopping List
            </ModalTitle>
          </ModalHeader>
          <ModalBody
            bsClass="modal-body"
            componentClass="div"
          >
            <form
              onSubmit={[Function]}
            >
              <FormGroup
                bsClass="form-group"
                controlId="Listname"
              >
                <ControlLabel
                  bsClass="control-label"
                  srOnly={false}
                >
                  Listname
                </ControlLabel>
                <FormControl
                  bsClass="form-control"
                  componentClass="input"
                  name="listname"
                  onChange={[Function]}
                  placeholder="Enter listname"
                  required={true}
                  type="text"
                  value={undefined}
                />
              </FormGroup>
              <Button
                active={false}
                block={false}
                bsClass="btn"
                bsStyle="default"
                disabled={false}
                type="submit"
              >
                Submit
              </Button>
              <FormControlFeedback
                bsClass="form-control-feedback"
                bsRole="feedback"
              />
            </form>
          </ModalBody>
          <ModalFooter
            bsClass="modal-footer"
            componentClass="div"
          >
            <Button
              active={false}
              block={false}
              bsClass="btn"
              bsStyle="default"
              disabled={false}
              onClick={[Function]}
            >
              Close
            </Button>
          </ModalFooter>
        </Modal>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "btn glyphicon glyphicon-edit text-primary",
          "data-placement": "top",
          "data-toggle": "tooltip",
          "onClick": [Function],
          "title": "Update_List",
          "type": "button",
        },
        "ref": null,
        "rendered": null,
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "animation": true,
          "autoFocus": true,
          "backdrop": true,
          "bsClass": "modal",
          "children": Array [
            <ModalHeader
              bsClass="modal-header"
              closeButton={true}
              closeLabel="Close"
            >
              <ModalTitle
                bsClass="modal-title"
                componentClass="h4"
              >
                Edit Shopping List
              </ModalTitle>
            </ModalHeader>,
            <ModalBody
              bsClass="modal-body"
              componentClass="div"
            >
              <form
                onSubmit={[Function]}
              >
                <FormGroup
                  bsClass="form-group"
                  controlId="Listname"
                >
                  <ControlLabel
                    bsClass="control-label"
                    srOnly={false}
                  >
                    Listname
                  </ControlLabel>
                  <FormControl
                    bsClass="form-control"
                    componentClass="input"
                    name="listname"
                    onChange={[Function]}
                    placeholder="Enter listname"
                    required={true}
                    type="text"
                    value={undefined}
                  />
                </FormGroup>
                <Button
                  active={false}
                  block={false}
                  bsClass="btn"
                  bsStyle="default"
                  disabled={false}
                  type="submit"
                >
                  Submit
                </Button>
                <FormControlFeedback
                  bsClass="form-control-feedback"
                  bsRole="feedback"
                />
              </form>
            </ModalBody>,
            <ModalFooter
              bsClass="modal-footer"
              componentClass="div"
            >
              <Button
                active={false}
                block={false}
                bsClass="btn"
                bsStyle="default"
                disabled={false}
                onClick={[Function]}
              >
                Close
              </Button>
            </ModalFooter>,
          ],
          "dialogComponentClass": [Function],
          "enforceFocus": true,
          "keyboard": true,
          "manager": ModalManager {
            "add": [Function],
            "containers": Array [],
            "data": Array [],
            "handleContainerOverflow": true,
            "hideSiblingNodes": true,
            "isTopModal": [Function],
            "modals": Array [],
            "remove": [Function],
          },
          "onHide": [Function],
          "renderBackdrop": [Function],
          "restoreFocus": true,
          "show": false,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "modal-header",
              "children": <ModalTitle
                bsClass="modal-title"
                componentClass="h4"
              >
                Edit Shopping List
              </ModalTitle>,
              "closeButton": true,
              "closeLabel": "Close",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "bsClass": "modal-title",
                "children": "Edit Shopping List",
                "componentClass": "h4",
              },
              "ref": null,
              "rendered": "Edit Shopping List",
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "modal-body",
              "children": <form
                onSubmit={[Function]}
              >
                <FormGroup
                  bsClass="form-group"
                  controlId="Listname"
                >
                  <ControlLabel
                    bsClass="control-label"
                    srOnly={false}
                  >
                    Listname
                  </ControlLabel>
                  <FormControl
                    bsClass="form-control"
                    componentClass="input"
                    name="listname"
                    onChange={[Function]}
                    placeholder="Enter listname"
                    required={true}
                    type="text"
                    value={undefined}
                  />
                </FormGroup>
                <Button
                  active={false}
                  block={false}
                  bsClass="btn"
                  bsStyle="default"
                  disabled={false}
                  type="submit"
                >
                  Submit
                </Button>
                <FormControlFeedback
                  bsClass="form-control-feedback"
                  bsRole="feedback"
                />
              </form>,
              "componentClass": "div",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <FormGroup
                    bsClass="form-group"
                    controlId="Listname"
                  >
                    <ControlLabel
                      bsClass="control-label"
                      srOnly={false}
                    >
                      Listname
                    </ControlLabel>
                    <FormControl
                      bsClass="form-control"
                      componentClass="input"
                      name="listname"
                      onChange={[Function]}
                      placeholder="Enter listname"
                      required={true}
                      type="text"
                      value={undefined}
                    />
                  </FormGroup>,
                  <Button
                    active={false}
                    block={false}
                    bsClass="btn"
                    bsStyle="default"
                    disabled={false}
                    type="submit"
                  >
                    Submit
                  </Button>,
                  <FormControlFeedback
                    bsClass="form-control-feedback"
                    bsRole="feedback"
                  />,
                ],
                "onSubmit": [Function],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "bsClass": "form-group",
                    "children": Array [
                      <ControlLabel
                        bsClass="control-label"
                        srOnly={false}
                      >
                        Listname
                      </ControlLabel>,
                      <FormControl
                        bsClass="form-control"
                        componentClass="input"
                        name="listname"
                        onChange={[Function]}
                        placeholder="Enter listname"
                        required={true}
                        type="text"
                        value={undefined}
                      />,
                    ],
                    "controlId": "Listname",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "bsClass": "control-label",
                        "children": "Listname",
                        "srOnly": false,
                      },
                      "ref": null,
                      "rendered": "Listname",
                      "type": [Function],
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "bsClass": "form-control",
                        "componentClass": "input",
                        "name": "listname",
                        "onChange": [Function],
                        "placeholder": "Enter listname",
                        "required": true,
                        "type": "text",
                        "value": undefined,
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                  ],
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "active": false,
                    "block": false,
                    "bsClass": "btn",
                    "bsStyle": "default",
                    "children": "Submit",
                    "disabled": false,
                    "type": "submit",
                  },
                  "ref": null,
                  "rendered": "Submit",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "bsClass": "form-control-feedback",
                    "bsRole": "feedback",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "form",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "modal-footer",
              "children": <Button
                active={false}
                block={false}
                bsClass="btn"
                bsStyle="default"
                disabled={false}
                onClick={[Function]}
              >
                Close
              </Button>,
              "componentClass": "div",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "active": false,
                "block": false,
                "bsClass": "btn",
                "bsStyle": "default",
                "children": "Close",
                "disabled": false,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Close",
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="btn glyphicon glyphicon-edit text-primary"
            data-placement="top"
            data-toggle="tooltip"
            onClick={[Function]}
            title="Update_List"
            type="button"
          />,
          <Modal
            animation={true}
            autoFocus={true}
            backdrop={true}
            bsClass="modal"
            dialogComponentClass={[Function]}
            enforceFocus={true}
            keyboard={true}
            manager={
              ModalManager {
                "add": [Function],
                "containers": Array [],
                "data": Array [],
                "handleContainerOverflow": true,
                "hideSiblingNodes": true,
                "isTopModal": [Function],
                "modals": Array [],
                "remove": [Function],
              }
            }
            onHide={[Function]}
            renderBackdrop={[Function]}
            restoreFocus={true}
            show={false}
          >
            <ModalHeader
              bsClass="modal-header"
              closeButton={true}
              closeLabel="Close"
            >
              <ModalTitle
                bsClass="modal-title"
                componentClass="h4"
              >
                Edit Shopping List
              </ModalTitle>
            </ModalHeader>
            <ModalBody
              bsClass="modal-body"
              componentClass="div"
            >
              <form
                onSubmit={[Function]}
              >
                <FormGroup
                  bsClass="form-group"
                  controlId="Listname"
                >
                  <ControlLabel
                    bsClass="control-label"
                    srOnly={false}
                  >
                    Listname
                  </ControlLabel>
                  <FormControl
                    bsClass="form-control"
                    componentClass="input"
                    name="listname"
                    onChange={[Function]}
                    placeholder="Enter listname"
                    required={true}
                    type="text"
                    value={undefined}
                  />
                </FormGroup>
                <Button
                  active={false}
                  block={false}
                  bsClass="btn"
                  bsStyle="default"
                  disabled={false}
                  type="submit"
                >
                  Submit
                </Button>
                <FormControlFeedback
                  bsClass="form-control-feedback"
                  bsRole="feedback"
                />
              </form>
            </ModalBody>
            <ModalFooter
              bsClass="modal-footer"
              componentClass="div"
            >
              <Button
                active={false}
                block={false}
                bsClass="btn"
                bsStyle="default"
                disabled={false}
                onClick={[Function]}
              >
                Close
              </Button>
            </ModalFooter>
          </Modal>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "btn glyphicon glyphicon-edit text-primary",
            "data-placement": "top",
            "data-toggle": "tooltip",
            "onClick": [Function],
            "title": "Update_List",
            "type": "button",
          },
          "ref": null,
          "rendered": null,
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "animation": true,
            "autoFocus": true,
            "backdrop": true,
            "bsClass": "modal",
            "children": Array [
              <ModalHeader
                bsClass="modal-header"
                closeButton={true}
                closeLabel="Close"
              >
                <ModalTitle
                  bsClass="modal-title"
                  componentClass="h4"
                >
                  Edit Shopping List
                </ModalTitle>
              </ModalHeader>,
              <ModalBody
                bsClass="modal-body"
                componentClass="div"
              >
                <form
                  onSubmit={[Function]}
                >
                  <FormGroup
                    bsClass="form-group"
                    controlId="Listname"
                  >
                    <ControlLabel
                      bsClass="control-label"
                      srOnly={false}
                    >
                      Listname
                    </ControlLabel>
                    <FormControl
                      bsClass="form-control"
                      componentClass="input"
                      name="listname"
                      onChange={[Function]}
                      placeholder="Enter listname"
                      required={true}
                      type="text"
                      value={undefined}
                    />
                  </FormGroup>
                  <Button
                    active={false}
                    block={false}
                    bsClass="btn"
                    bsStyle="default"
                    disabled={false}
                    type="submit"
                  >
                    Submit
                  </Button>
                  <FormControlFeedback
                    bsClass="form-control-feedback"
                    bsRole="feedback"
                  />
                </form>
              </ModalBody>,
              <ModalFooter
                bsClass="modal-footer"
                componentClass="div"
              >
                <Button
                  active={false}
                  block={false}
                  bsClass="btn"
                  bsStyle="default"
                  disabled={false}
                  onClick={[Function]}
                >
                  Close
                </Button>
              </ModalFooter>,
            ],
            "dialogComponentClass": [Function],
            "enforceFocus": true,
            "keyboard": true,
            "manager": ModalManager {
              "add": [Function],
              "containers": Array [],
              "data": Array [],
              "handleContainerOverflow": true,
              "hideSiblingNodes": true,
              "isTopModal": [Function],
              "modals": Array [],
              "remove": [Function],
            },
            "onHide": [Function],
            "renderBackdrop": [Function],
            "restoreFocus": true,
            "show": false,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "bsClass": "modal-header",
                "children": <ModalTitle
                  bsClass="modal-title"
                  componentClass="h4"
                >
                  Edit Shopping List
                </ModalTitle>,
                "closeButton": true,
                "closeLabel": "Close",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "bsClass": "modal-title",
                  "children": "Edit Shopping List",
                  "componentClass": "h4",
                },
                "ref": null,
                "rendered": "Edit Shopping List",
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "bsClass": "modal-body",
                "children": <form
                  onSubmit={[Function]}
                >
                  <FormGroup
                    bsClass="form-group"
                    controlId="Listname"
                  >
                    <ControlLabel
                      bsClass="control-label"
                      srOnly={false}
                    >
                      Listname
                    </ControlLabel>
                    <FormControl
                      bsClass="form-control"
                      componentClass="input"
                      name="listname"
                      onChange={[Function]}
                      placeholder="Enter listname"
                      required={true}
                      type="text"
                      value={undefined}
                    />
                  </FormGroup>
                  <Button
                    active={false}
                    block={false}
                    bsClass="btn"
                    bsStyle="default"
                    disabled={false}
                    type="submit"
                  >
                    Submit
                  </Button>
                  <FormControlFeedback
                    bsClass="form-control-feedback"
                    bsRole="feedback"
                  />
                </form>,
                "componentClass": "div",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <FormGroup
                      bsClass="form-group"
                      controlId="Listname"
                    >
                      <ControlLabel
                        bsClass="control-label"
                        srOnly={false}
                      >
                        Listname
                      </ControlLabel>
                      <FormControl
                        bsClass="form-control"
                        componentClass="input"
                        name="listname"
                        onChange={[Function]}
                        placeholder="Enter listname"
                        required={true}
                        type="text"
                        value={undefined}
                      />
                    </FormGroup>,
                    <Button
                      active={false}
                      block={false}
                      bsClass="btn"
                      bsStyle="default"
                      disabled={false}
                      type="submit"
                    >
                      Submit
                    </Button>,
                    <FormControlFeedback
                      bsClass="form-control-feedback"
                      bsRole="feedback"
                    />,
                  ],
                  "onSubmit": [Function],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "bsClass": "form-group",
                      "children": Array [
                        <ControlLabel
                          bsClass="control-label"
                          srOnly={false}
                        >
                          Listname
                        </ControlLabel>,
                        <FormControl
                          bsClass="form-control"
                          componentClass="input"
                          name="listname"
                          onChange={[Function]}
                          placeholder="Enter listname"
                          required={true}
                          type="text"
                          value={undefined}
                        />,
                      ],
                      "controlId": "Listname",
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "class",
                        "props": Object {
                          "bsClass": "control-label",
                          "children": "Listname",
                          "srOnly": false,
                        },
                        "ref": null,
                        "rendered": "Listname",
                        "type": [Function],
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "class",
                        "props": Object {
                          "bsClass": "form-control",
                          "componentClass": "input",
                          "name": "listname",
                          "onChange": [Function],
                          "placeholder": "Enter listname",
                          "required": true,
                          "type": "text",
                          "value": undefined,
                        },
                        "ref": null,
                        "rendered": null,
                        "type": [Function],
                      },
                    ],
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "active": false,
                      "block": false,
                      "bsClass": "btn",
                      "bsStyle": "default",
                      "children": "Submit",
                      "disabled": false,
                      "type": "submit",
                    },
                    "ref": null,
                    "rendered": "Submit",
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "bsClass": "form-control-feedback",
                      "bsRole": "feedback",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                ],
                "type": "form",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "bsClass": "modal-footer",
                "children": <Button
                  active={false}
                  block={false}
                  bsClass="btn"
                  bsStyle="default"
                  disabled={false}
                  onClick={[Function]}
                >
                  Close
                </Button>,
                "componentClass": "div",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "active": false,
                  "block": false,
                  "bsClass": "btn",
                  "bsStyle": "default",
                  "children": "Close",
                  "disabled": false,
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Close",
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
